discord_connector.py

```diff
@@ -413,6 +466,70 @@ def run_in_loop(self, coroutine, action_result, message=""):
             return action_result.set_status(phantom.APP_ERROR,
                                             f"Other exception. Error type: {e.__class__.__name__} Details: {str(e)}"), None
 
+    def _handle_on_poll(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        if not self.is_poll_now():
+            self.debug_print("Maximum number of containers have been reached")
+            return action_result.set_status(phantom.APP_ERROR, "Maximum number of containers have been reached")
+
+        container_count = param["container_count"]
+
+        self._status = self.load_state()
+        last_poll = self._state.get("last poll", None)
+        if last_poll is not None:
+            _, last_poll = self.parse_date(last_poll)
+        self._state["last poll"] = str(datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        self.save_state(self._state)
+
+        status, channels = self.run_in_loop(self._guild.fetch_channels(), action_result,
+                                            message="Cannot fetch channel from Discord.")
+        if not status:
+            self.debug_print("action result: Cannot Poll messages, unable to fetch channels ")
+            return action_result.set_status(phantom.APP_ERROR, "action result: Cannot Poll messages")
+
+        for channel in channels:
+            if isinstance(channel, discord.TextChannel):
```

Comment:
I'd try to do something to not encapsulate all for body in if statement

Reply:
It has been chopped to two separated methods.